src/mon/PGMonitor.cc

```diff
@@ -190,6 +190,8 @@ void PGMonitor::update_from_paxos(bool *need_bootstrap)
     int r = get_version(pg_map.version + 1, bl);
     if (r == -ENOENT) {
       dout(10) << __func__ << " failed to read_incremental, read_full" << dendl;
+      // reset pg map
+      pg_map = PGMap();
```

Comment:
@tchaikov yeah, we assert on `out`. Missed that. As for not being the best way, when we get to the point of a full version, or an incremental, being missing, then something went wrong with paxos at some point (sync, recovery, or proposal/commit), and asserting out doesn't seem like the wrong way to go. Would be nicer to have a self-healing attempt round though, but in the absence of that I'd much rather the monitor die a horrible death than having an inconsistent store.